### PR TITLE
feat: add bdd test for vault and split the setup from bdd execution

### PR DESCRIPTION
**Pull request by ccojocar**
part of jenkins-x/jx#5190

### CODE REVIEW

File: jenkins-x-bdd-vault.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                memory: 2048Mi
+        environment:
+          - name: GOPROXY
+            value: http://jenkins-x-athens-proxy:80
+        agent:
+          image: gcr.io/jenkinsxio/builder-go-maven:0.1.645
+        stages:
+          - name: ci
+            steps:
+              - comamnd: bdd/boot-vault/setup.sh
```
Comment (abayer) [inline jenkins-x-bdd-vault.yml]: `comamnd` is thoroughly not right. =)

File: jenkins-x-bdd-vault.yml [hunk 2]
```diff
@@ -21,7 +21,9 @@ pipelineConfig:
           - name: ci
             steps:
               - name: setup
-                command: bdd/boot-vault/setup.sh
+                command: bdd/setup.sh
+                args:['bdd/boot-vault', 'bdd-config']
```
Comment (abayer) [inline jenkins-x-bdd-vault.yml]: Needs to be a space after the `:`- same in `jenkins-x-bdd-local.yml`.

Comment (ccojocar) [reply jenkins-x-bdd-vault.yml]: yap. I think we need to improve the message return when the pipeline runner fails

Comment (abayer) [reply jenkins-x-bdd-vault.yml]: Metapipeline helps with that to some extent.

**Pull request merged**